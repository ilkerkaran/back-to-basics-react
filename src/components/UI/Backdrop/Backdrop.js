import React from 'react';
import PropTypes from 'prop-types';
import './Backdrop.css';

const backdrop = props => {
  return props.show ? (
    <div className="Backdrop" onClick={props.onClick}>
      {' '}
    </div>
  ) : null;
};

export default backdrop;

backdrop.propTypes = {
  show: PropTypes.bool.isRequired,
  onClick: PropTypes.func.isRequired
};
