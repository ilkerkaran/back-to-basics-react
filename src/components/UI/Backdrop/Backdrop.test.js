import React from 'react';
import renderer from 'react-test-renderer';
import Backdrop from './Backdrop';


describe('<Backdrop /> snapshot ', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Backdrop show onClick={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
