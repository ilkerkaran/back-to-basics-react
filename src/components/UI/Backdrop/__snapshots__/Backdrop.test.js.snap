// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Backdrop /> snapshot  renders correctly 1`] = `
<div
  className="Backdrop"
  onClick={[Function]}
>
   
</div>
`;
